clang/tools/clang-scan-deps/clang-deps-launcher.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+"""
+Launcher for clang which computes the dependency for the file being compiled.
+"""
+import sys
+import os
+import subprocess
+import time
+import json
+import tempfile
+
+
+TOOLS_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+def trim_clang_args_in_cdb(cdb):
+    """Trim -Xclang options away since they are not portable"""
+    # We also replace the clang exe path so it can find the clang headers.
+    clang_exe = os.path.basename(cdb['arguments'][0])
+    trim_args = [os.path.join(TOOLS_DIR, clang_exe)]
+    skip = False
+    for arg in cdb['arguments'][1:]:
+        if skip:
+            skip = False
+            continue
+        if arg == '-Xclang':
+            skip = True
+            continue
+        trim_args.append(arg)
+    cdb['arguments'] = trim_args
+
+
+def main():
+    # Clang command is the input.
+    clang_cmd = sys.argv[1:]
+    # Figure out the output path.
+    output_idx = len(clang_cmd) - clang_cmd[::-1].index('-o')
```

Comment:
Error if there is no explicit output?

Reply:
You mean no `-o` output? That is an exception. There is no exception handling currently and all of them will error out and treated as error.

Reply:
Oh, I was thinking of "str.find" that returns -1.  I guess `ValueError: '-o' is not in list` is good enough for now.